### PR TITLE
Integrate Endpoints for Posture Checks

### PR DESCRIPTION
**Pull request by bcmmbaga**
## Describe your changes

 Introduces a fundamental endpoint for handling posture checks, enabling the functionality to create, read, update, and delete posture checks.

Create a Posture Check
```
curl --request POST \
  --url http://localhost:33073/api/posture-checks \
  --header 'Accept: application/json' \
  --header 'Authorization: Token <PAT>' \
  --header 'Content-Type: application/json' \
  --data '{
	"name": "Default",
	"description": "This checks if the peer is running the required NetBird's version",
	"checks": {
		"nb_version_check": {
			"min_version": "0.25.0"
		}
	}
}'
```

List all Posture Checks
```
curl --request GET \
  --url http://localhost:33073/api/posture-checks \
  --header 'Accept: application/json' \
  --header 'Authorization: Token <PAT>'
```
Get Posture Check
```
curl --request GET \
  --url http://localhost:33073/api/posture-checks/<ID>\
  --header 'Accept: application/json' \
  --header 'Authorization: Token <PAT>'
```

Update Posture Check
```
curl --request PUT \
  --url http://localhost:33073/api/posture-checks/<ID>\
  --header 'Accept: application/json' \
  --header 'Authorization: Token <PAT>' \
  --header 'Content-Type: application/json' \
  --data '{
	"name": "Default",
	"description": "This checks if the peer is running required NetBird's version",
	"checks": {
		"nb_version_check": {
			"min_version": "1.2.1"
		}
	}
}'
```

Delete Posture Check
```
curl --request DELETE \
  --url http://localhost:33073/api/posture-checks/<ID>\
  --header 'Accept: application/json' \
  --header 'Authorization: Token <PAT>'
```
  
  
### Checklist
- [ ] Is it a bug fix
- [ ] Is a typo/documentation fix
- [x] Is a feature enhancement
- [ ] It is a refactor
- [ ] Created tests that fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: management/server/posture/checks.go [hunk 1]
```diff
@@ -43,3 +50,67 @@ func (pc *Checks) Copy() *Checks {
...
+		result[check.Name()] = data
+	}
+	return result
+}
+
+func (pc *Checks) unmarshalChecks(rawChecks map[string]json.RawMessage) error {
+	pc.Checks = make([]Check, 0, len(rawChecks))
+
+	for name, rawCheck := range rawChecks {
+		switch name {
```
Comment (surik) [inline management/server/posture/checks.go:107]: You can add `//nolint:gocritic` to make linter happy here

File: management/server/http/api/openapi.yml [hunk 2]
```diff
@@ -791,13 +791,78 @@ components:
...
+      type: object
+      properties:
+        enabled:
+          description: NetBird's version check status
+          type: boolean
+          example: false
+        min_version:
+          description: Minimum acceptable NetBird version
+          type: string
+          example: "1.2.3"
```
Comment (surik) [inline management/server/http/api/openapi.yml]: Maybe put there a more realistic version like 0.25.0

File: management/server/posture/version_test.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+)
+
+func TestNBVersionCheck_Check(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   peer.Peer
+		check   NBVersionCheck
+		wantErr bool
+	}{
+		{
```
Comment (surik) [inline management/server/posture/version_test.go:18]: I suggest we extend tests to the following cases:
1. When `MinVersion` is `2.0` and `UIVersion` is `2.0.0` get no error
2. When `MinVersion` is `0.2` and `UIVersion` is `0.1.0` get an error
3. When `MinVersion` is `0.2.1` and `UIVersion` is `0.2.1-test` get no error
4. When `MinVersion` is `0.2.1` and `UIVersion` is `0.2.0-test` get an error

Comment (bcmmbaga) [reply management/server/posture/version_test.go:18]: Regarding test cases 3 and 4, the [go-version package](https://github.com/hashicorp/go-version) treats these versions differently. A constraint with a pre-release can only match a pre-release version with the same base segments, and similarly, a constraint without a pre-release can only match a version without a pre-release

File: management/server/posture/version.go [hunk 4]
```diff
@@ -11,19 +11,21 @@ import (
...
 }
 
 var _ Check = (*NBVersionCheck)(nil)
 
 func (n *NBVersionCheck) Check(peer nbpeer.Peer) error {
+	if !n.Enabled {
+		return nil
+	}
+
 	peerNBVersion, err := version.NewVersion(peer.Meta.UIVersion)
```
Comment (surik) [inline management/server/posture/version.go]: We should compare it with `peer.Meta.WtVersion`

**Review by surik — APPROVED** [review]

**Pull request merged**